File: my_project_typescript/medium_problems_typescript/from1to50/mergeIntervals.ts
function merge(intervals: number[][]): number[][] {

    let out: Array<Array<number>> = []

    intervals = intervals.sort( (a,b) => a[0]<=b[0]?-1:1)

    for (let i of intervals){
        if (out.length > 0 && (i[0] <= out[out.length -1][out[out.length -1].length-1])){
            out[out.length -1][out[out.length -1].length-1] = Math.max(out[out.length -1][out[out.length -1].length-1], i[i.length-1])
        }
        else {
            out = out.concat(Array(i))
        }
    }

    console.log(intervals)

    return out 
};


console.log(merge([[2,6],[8,10],[1,3],[15,18]]))

File: src/my_project/interviews/amazon_high_frequency_23/round_2/merge_intervals.py
from typing import List, Union, Collection, Mapping, Optional
from abc import ABC, abstractmethod

class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:

        list_answer = list()

        for i in sorted(intervals, key=lambda i: i[0]):

            if list_answer and i[0] <= list_answer[-1][-1]:
                list_answer[-1][-1] = max(i[1], list_answer[-1][-1])
            else: 
                list_answer.append(i)

        return list_answer